src/imars3dv2/filters/tilt.py

```diff
@@ -0,0 +1,301 @@
+        The rotation axis tilt angle in degrees
+    @param ncore:
+        Number of cores to use for parallel median filtering, default is -1, which means using all available cores.
+
+    Returns
+    -------
+    @return:
+        The tilt corrected array
+    """
+    # dimensionality check
+    if arrays.ndim == 2:
+        return rotate(arrays, -tilt, resize=False, preserve_range=True)
+    elif arrays.ndim == 3:
+
+        def tilt_correct_image(idx):
+            return rotate(arrays[idx], -tilt, resize=False, preserve_range=True)
+
+        # NOTE: have to switch to threadpool as we are using a nested function
+        ncore = mproc.mp.cpu_count() if ncore == -1 else int(ncore)
+        with cf.ThreadPoolExecutor(ncore) as e:
+            jobs = [e.submit(tilt_correct_image, idx) for idx in range(arrays.shape[0])]
+        arrays = np.array([job.result() for job in jobs])
+        return arrays
+    else:
+        raise ValueError("array must be a 2d/3d array")
+
+
+def find_180_deg_pairs_idx(
+    angles: np.ndarray,
+    atol: float = 1e-3,
+    in_degrees: bool = True,
+) -> Tuple[np.ndarray, np.ndarray]:
+    """
+    Return the indices of the 180 degree pairs from given list of angles.
+
+    Parameters
+    ----------
+    @param angles:
+        The list of angles as a 1d array.
+    @param atol:
+        The absolute tolerance in degree for the 180 degree pairs.
+    @param in_degrees:
+        Whether the angles are in degrees or radians, default is in degrees.
+
+    Returns
+    -------
+    @return:
+        The indices of the 180 degree pairs (low_range, high_range)
+    """
+    # ensure correct dimension
+    if angles.ndim != 1:
+        raise ValueError("angles must be a 1d array")
+    # ensure angles are in degrees
+    angles = angles if in_degrees else np.degrees(angles)
+    # compute the self difference matrix
+    angles = angles[..., np.newaxis]
+    diff_matrix = angles.T - angles
+    # find the indices of the 180 degree pairs
+    idx_lowrange, idx_highrange = np.where(np.isclose(diff_matrix, 180, atol=atol))
+    return idx_lowrange, idx_highrange
+
+
+def calculate_shift(
+    reference_image: np.ndarray,
+    moving_image: np.ndarray,
+) -> float:
+    """
+    Calculate the amount of shift needed to move the moving image to the
+    reference image.
+    This method requires the rotating object to be fully in the field of view.
+
+    Parameters
+    ----------
+    @param reference_image:
+        The reference image, often the radiograph taken at omega (< 180 deg)
+    @param moving_image:
+        The moving image, often the radiograph taken at omega + 180 deg
+
+    Returns
+    -------
+    @return:
+        The amount of shift in pixels
+    """
+    #
+    shift = phase_cross_correlation(
+        reference_image=reference_image,
+        moving_image=np.fliplr(moving_image),
+        upsample_factor=2.0,  # upsample by 2x for subpixel accuracy
+    )
+    #
+    return shift[0][1]
```

Comment:
Would it be good to document why this returns this particular index? It feels like a magic value

Reply:
additional information is added to explain the usage of `phase_cross_correlation`